x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/MlIndexAndAlias.java

```diff
@@ -69,13 +72,38 @@ private MlIndexAndAlias() {}
      * Adds an {@code alias} to that index if it was created,
      * or to the index with the highest suffix if the index did not have to be created.
      * The listener is notified with a {@code boolean} that informs whether the index or the alias were created.
+     * If the index is created, the listener is not called until the index is ready to use via the supplied alias,
+     * so that a method that receives a success response from this method can safely use the index immediately.
      */
     public static void createIndexAndAliasIfNecessary(Client client,
                                                       ClusterState clusterState,
                                                       IndexNameExpressionResolver resolver,
                                                       String indexPatternPrefix,
                                                       String alias,
-                                                      ActionListener<Boolean> listener) {
+                                                      ActionListener<Boolean> finalListener) {
+
+        // If the index and alias were successfully created then wait for the shards of the index that the alias points to be ready
+        ActionListener<Boolean> waitForShardsListener = ActionListener.wrap(
```

Comment:
TBH it is always unclear to me how should I name the listener variable: after the action that takes place *inside* the listener or after the action that invokes the listener in the end?
Is there any codebase-wide guideline for that?